account_move_tree_purchase_date_origin/models/account_move.py

```diff
@@ -0,0 +1,21 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import fields, models, api
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    purchase_date_origin = fields.Datetime(compute="_compute_purchase_date_origin")
+
+    @api.depends('invoice_line_ids')
```

Comment:
Maybe more accurate (not sure):
```suggestion
    @api.depends('invoice_line_ids.purchase_line_id.order_id.date_order')
```